stackslib/src/net/inv/epoch2x.rs

```diff
@@ -2623,6 +2623,216 @@ impl PeerNetwork {
+        } else {
+            // do we have an always-allowed peer that we have not fully synced
+            // with?
+            let mut have_unsynced = false;
+            if let Some(ref inv_state) = self.inv_state {
+                for (nk, stats) in inv_state.block_stats.iter() {
+                    if self.is_bound(&nk) {
+                        // this is the same address we're bound to
+                        continue;
+                    }
+                    if Some((nk.addrbytes.clone(), nk.port)) == self.local_peer.public_ip_address {
+                        // this is a peer at our address
+                        continue;
+                    }
+                    if !always_allowed.contains(&nk) {
+                        // this peer isn't in the always-allowed set
+                        continue;
+                    }
+
+                    if stats.inv.num_reward_cycles
+                        >= self.pox_id.num_inventory_reward_cycles() as u64
+                    {
+                        // we have fully sync'ed with an always-allowed peer
+                        debug!(
+                            "{:?}: Fully-sync'ed PoX inventory from {}",
+                            self.get_local_peer(),
+                            nk,
+                        );
+                        finished_always_allowed_inv_sync = true;
+                    } else {
+                        // there exists an always-allowed peer that we have not
+                        // fully sync'ed with
+                        debug!(
+                            "{:?}: Have not fully sync'ed with {}",
+                            self.get_local_peer(),
+                            nk,
+                        );
+                        have_unsynced = true;
+                    }
+                }
+            }
+
+            if !have_unsynced {
+                // There exists one or more always-allowed peers in
+                // the inv state machine (per the peer DB), but all such peers
+                // report either our bind address or our public IP address.
+                // If this is the case (i.e. a configuration error, a weird
+                // case where nodes share an IP, etc), then we declare this inv
+                // sync pass as finished.
+                finished_always_allowed_inv_sync = true;
+            }
+        }
+
+        finished_always_allowed_inv_sync
+    }
+
+    /// Do an inventory state machine pass for epoch 2.x.
+    /// Returns the new work state  
+    pub fn work_inv_sync_epoch2x(
+        &mut self,
+        sortdb: &SortitionDB,
+        download_backpressure: bool,
+        ibd: bool,
+    ) -> PeerNetworkWorkState {
+        let mut work_state = PeerNetworkWorkState::BlockInvSync;
+
+        // synchronize epcoh 2.x peer block inventories
+        let (inv_done, inv_throttled) = self.do_network_inv_sync_epoch2x(sortdb, ibd);
+        if inv_done {
```

Comment:
Sure; again, this is just what was already there.  Only mentioning it because I'm trying to save you and @jbencin the time of having to review soon-to-be-legacy code that's been in production for years already ;)

Reply:
Yes, but I think it's a good opportunity to do simple cleanup as long as the behaviors are preserved.